openssl/ecdsa.go

```diff
@@ -67,54 +79,66 @@ func NewPublicKeyECDSA(curve string, X, Y *big.Int) (*PublicKeyECDSA, error) {
 	return k, nil
 }
 
-func newECKey(curve string, X, Y *big.Int) (*C.EC_KEY, error) {
-	nid, err := curveNID(curve)
-	if err != nil {
+func newECKey(curve string, X, Y, D *big.Int) (pkey *C.EVP_PKEY, err error) {
+	var nid C.int
+	if nid, err = curveNID(curve); err != nil {
```

Comment:
Is it purely a stylistic choice to combine the two lines? Is there a general precedent or style on when to do it?

Reply:
When the variables assigned to the function result are already declared, as in this case, the choice is purely stylistic. When I was writing that code I felt like it would look better, although I'm not that sure now 😝.

Reply:
I've seen this style pop up a few times, usually in the case when the error variable is declared though. That is the style shown in the Golang reference. 

I'm not against this style, I just don't have a good intuition as to when it should / shouldn't be used.

Reply:
Shadowing seems particularly troublesome in this case because it's shadowing a named return that is used in `defer`. There are special rules for `defer` and named returns. Could be very easy to miss the shadowing here and think you're in the named return mode for `defer`